### ISSUE TITLE
FIFA STREET [BLES01493] Audio is negatively impacted by Write Colour Buffers

### ISSUE THREAD

**Issue by ItsLaguna** [RPCS3/rpcs3#16371]
### Quick summary

FIFA STREET needs Write Colour Buffers to fix inaccurate colours on the crowd and bugged character portraits of players in the World Tour mode, but if this option is enabled then the audio is affected negatively as it will crackle, distort and sometimes slow down.

Here's gameplay without Write Colour Buffers to showcase the inaccurate colours of the crowd but the lack of crackling:
[FIFA Street w/o Write Colour Buffers](https://streamable.com/kvgpwf)

And here's gameplay with Write Colour Buffers enabled, which fixes the crowd but introduces the crackling:
[FIFA Street w/ Write Colour Buffers](https://streamable.com/o9lvnp)

I'm also mentioning the following issue as it's similar #15649 though not as bad as it is in this game.

### Attach a log file

[fifastreetwcb.log](https://github.com/user-attachments/files/18010123/fifastreetwcb.log)
[fifastreet no wcb.log](https://github.com/user-attachments/files/18010120/fifastreet.no.wcb.log)

### System configuration

Bazzite 41 (Fedora Atomic)
Ryzen 7 5800X
RTX 2060 | 565.57.1.0

**Comment by Virtua64** [comment]
an easy fix is to change the PPU thread count to 3 or 4

**Comment by ItsLaguna** [comment]
> an easy fix is to change the PPU thread count to 3 or 4

This doesn't do anything...

**Comment by kd-11** [comment]
Try using the build from https://github.com/RPCS3/rpcs3/pull/16352.
There really isn't anything much we can do about slowdown caused by GPU transfers. You need to increase your audio buffering window to compensate and maybe enable audio timestretch with threshold of like 95%. When WCB is enabled we have to wait for GPU to complete work (hard stall). This can happen in the context of a CELL thread which then causes the audio stuttering.

**Comment by ItsLaguna** [comment]
> Try using the build from #16352. There really isn't anything much we can do about slowdown caused by GPU transfers. You need to increase your audio buffering window to compensate and maybe enable audio timestretch with threshold of like 95%. When WCB is enabled we have to wait for GPU to complete work (hard stall). This can happen in the context of a CELL thread which then causes the audio stuttering.

I'm afraid that even in that build, with audio buffer set to 200ms and timestretch set to 95% the crackling is still terrible.
Not only that, but now write colour buffers doesn't fix the graphics it's supposed to fix.

[RPCS3.log](https://github.com/user-attachments/files/18030117/RPCS3.log)
[Video](https://streamable.com/qn14fh)

**Comment by Virtua64** [comment]
> > an easy fix is to change the PPU thread count to 3 or 4
> 
> This doesn't do anything...

Fixes it for me with a Ryzen 3700.  But YMMV

**Comment by kd-11** [comment]
Audio buffering is borked on linux though afaik, so that's probably not going to work. Someone broke it like 2 years ago when adding FAudio and it never got fixed.

**Comment by ItsLaguna** [comment]
Yeah, so far setting by it to PS3 Native (and disabling buffering) I haven't been able to get any crackling.
Waiting on someone to see if they can confirm that PS3 Native indeed fixes it, just to be sure.

I'll close the issue whenever that happens.

**Comment by Satan86** [comment]
Hello. I can confirm: setting Frame Limit to PS3 Native FIXES audio crackling on this game.

**Comment by ItsLaguna** [comment]
Closing as a fix was found.
